src/dataset_miner/data_extractor.py

```
@@ -0,0 +1,217 @@
+            except UnicodeDecodeError:
+                continue
+
+        if df is None:
+            raise TextExtractionError("Failed to read CSV with any supported encoding")
+
+        # Convert to string with formatting
+        text = df.to_string(index=False, na_rep="N/A")
+        logger.info(f"Extracted {len(text)} characters from {file_path}")
+        return text
+    except Exception as e:
+        raise TextExtractionError(f"Failed to extract text from CSV: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)